analysis/quantifier-comparisons/13-spatial-transcriptomics-benchmarking.Rmd

```diff
@@ -0,0 +1,411 @@
+```
+
+```{r}
+# total genes/ spot 
+ggplot(coldata_df_common, aes(x = detected, color = tool)) + 
+  geom_density() + 
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Genes detected/spot") + 
+  xlab("") 
+```
+```{r}
+plot_fry_spe(all_spe_filter$`alevin-fry`, "alevin-fry", "detected")
+```
+
+```{r}
+plot_spaceranger_spe(all_spe_filter$spaceranger, "spaceranger", "detected")
+```
+Generally it looks like both tools are fairly similar in the number of UMI/cell and genes detected/cell.
+
+### Per Gene QC Metrics
+
+Let's also look at the correlation of mean gene expression across shared genes. 
+We will first need to calculate the per feature QC on the filtered spes after removing spots not present in the tissue and then grab the `rowData` and combine into a data frame used for plotting. 
+
+```{r}
+all_spe_filter <- all_spe_filter %>%
+  purrr::map(scuttle::addPerFeatureQCMetrics)
+```
+
+```{r}
+# grab rowdata and combine with sample info
+rowdata_df <- purrr::map_df(all_spe_filter, spatial_rowdata_to_df, .id = "tool") %>%
+  dplyr::left_join(sample_info_df,
+                   by = c("tool"))
+```
+
+We then want to filter out any lowly detected genes, (detected < 5.0) and restrict our analysis to those genes that are found in both tools. 
+
+```{r}
+gene_counts <- rowdata_df %>% 
+  # remove genes that have a low frequency of being detected
+  dplyr::filter(detected >= 5.0) %>%
+  dplyr::count(gene_id, sample)
+
+# restrict to only common genes 
+common_genes <- gene_counts %>%
+  dplyr::filter(n == 2) %>%
+  dplyr::pull(gene_id)
+
+rowdata_df_common <- rowdata_df %>%
+  dplyr::filter(
+    (gene_id %in% common_genes) 
+  )
+```
+
+```{r}
+# create a table to calculate correlation between mean gene expression
+rowdata_cor <- rowdata_df_common %>%
+  dplyr::select(tool, gene_id, sample, mean) %>%
+  # spread the mean expression stats to one column per caller
+  tidyr::pivot_wider(id_cols = c(gene_id, sample),
+                     names_from = c("tool"),
+                     values_from = mean) %>%
+  # drop rows with NA values to ease correlation calculation
+  tidyr::drop_na()
+```
+
+```{r}
+# look at correlation between the two tools
+rowdata_cor %>% 
+  dplyr::group_by(sample) %>%
+  dplyr::summarize(
+    alevin_fry_knee_spaceranger_cor = cor(`spaceranger`, `alevin-fry`, method = "spearman")
+  )
+```
+
+```{r}
+# mean gene expression across shared genes 
+ggplot(rowdata_cor, aes(x = `spaceranger`, y = `alevin-fry`)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  labs(x = "Spaceranger mean gene expression", y = "Alevin Fry Mean gene expression") + 
+  theme_classic()
+```
+Correlation appears to be quite high between mean gene expression in Spaceranger and Alevin-fry, however, we do see that there is a subset of genes that have higher gene expression in Spaceranger than in Alevin-fry and are slightly off the diagonal. 
```

Comment:
Do we have any sense of what these genes are? It seems like a good number, and it is striking how it looks like there are two lines there...